### PR TITLE
Wip require action feature fail post

### PR DESCRIPTION
**Pull request by AvengerMoJo**
Still a WIP branch and try to implement the #26666 

In the earlier commit https://github.com/go-gitea/gitea/compare/main...AvengerMoJo:gitea:wip_require_action_feature 
I create a table to allow repo workflow to enable as global (org) visible action. 
However when I create a form to POST call to try creating the table to keep track of all the require action. Either the repo/setting/actions/require_action/add or repo/setting/actions/require_action in routers/web/web.go able to work. I get 405 or bad request. 
Need some tips to connect the bot for my POST request, 
web/repo/setting/actions/require_action interface 
from templates/shared/actions/require_action_list.tmpl click the add button calling 
routes/web/repo/setting/require_action
RequireActionsCreate call shared action
then 
from service/actions/require_action 
CreateRequireAction call InsertRequireAction in 
the model/actions/require_action

### CODE REVIEW

**Comment by AvengerMoJo** [comment]
2024/04/06 14:19:45 ...eb/routing/logger.go:102:func1() [I] router: completed POST /org/SUSE/settings/actions/require_action/add for [::1]:62841, 400 Bad Request in 9.4ms @ setting/require_action.go:73(setting.RequireActionCreate)

**Comment by silverwind** [comment]
Please fix all the lint errors and run `make fmt` :)

**Comment by AvengerMoJo** [comment]
@silverwind I fixed up the fmt and added the delete function. So currently you can enable a workflow in the repo to the org global level. Then you can add it into the RequireAction table in the Org->Setting->Actions->RequireAction. 
However, I am still missing how to assign them back to all the repo and schedule the runner then the action happen.

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -3640,11 +3640,38 @@ runs.no_workflows.documentation = For more information on Gitea Actions, see <a
...
+require_action.add_require_action = Enable selected Workflow
+require_action.new = Create New
+require_action.status = Status
+require_action.search = Search...
+require_action.version = Version
+require_action.repo = Repo Name
+require_action.workflow = Workflow Filename
+require_action.link = Link
+require_action.remove = Remove
+require_action.none = No Require Actions Available.
```
Comment (dirkmueller) [inline options/locale/locale_en-US.ini:3664]:
```suggestion
require_action.none = No Required Actions Available.
```

File: templates/shared/actions/require_action_list.tmpl [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+				checkboxes.forEach(function(otherCheckbox) {
+					var otherHiddenInput = otherCheckbox.nextElementSibling;
+					if (otherCheckbox !== checkbox) {
+						otherHiddenInput.disabled = isChecked;
+					}
+				});
+			});
+		});
+	});
+</script>
```
Comment (silverwind) [inline templates/shared/actions/require_action_list.tmpl:146]: This script should be in `web_src/js` and if necessary detect with a fast selector like `document.getElementById('add-require-actions-modal')` whether this modal is present, and if not abort.

Comment (AvengerMoJo) [reply templates/shared/actions/require_action_list.tmpl:146]: @silverwind does the following make sense? I'm not sure it should be in features or components.  
I am also wondering there is any default function allows me to disable and enable the rest of the input based on the selected radio button like what I am trying to do.

Comment (silverwind) [reply templates/shared/actions/require_action_list.tmpl:146]: `features` is fine, we pile everything into it currently.

Regarding the other question: I'm not sure but please check how it's done on the repo settings page. There is some mechanism there that enables/disables a bunch of fields when radio buttons are clicked.

**Comment by AvengerMoJo** [comment]
@silverwind I may need a bit help with the way global require action implementation after being enable . 
e.g routers/web/repo/actions/actions.go in List
Should I implement the global actions.ListGlobalWorkflows like the actions.ListWorkflows? ( I have  a hard time including the global data model in the actions/workflow.go ) 
Or should i follow the runners using db.Find[actions_models.RequireAction]

**Comment by AvengerMoJo** [comment]
https://github.com/go-gitea/gitea/compare/main...AvengerMoJo:gitea:wip_require_action

@silverwind  if you could try the above wip branch has a implementation of the RequireAction. The usage and screen capture of steps are located #26666. 
Thanks

**Comment by AvengerMoJo** [comment]
#31869 replace this PR

**Pull request closed without merging**